src/ape/api/providers.py

```diff
@@ -268,69 +339,162 @@ class ProviderAPI:
 
     @abstractmethod
     def connect(self):
-        ...
+        """
+        Connect a contract to a provider.
+        """
 
     @abstractmethod
     def disconnect(self):
-        ...
+        """
+        Disconnect a contract from a provider.
+        """
 
     @abstractmethod
     def update_settings(self, new_settings: dict):
-        ...
+        """
+        Change the settings connection protocol.
+        May require a reconnect.
+
+        Agrs:
+            new_settings ([`dict`]): Value of new provider
+        """
 
     @property
     @abstractmethod
     def chain_id(self) -> int:
-        ...
+        """
+        Gives value of blockchain id.
+
+        Returns:
+            int: value of blockchain id."""
 
     @abstractmethod
     def get_balance(self, address: str) -> int:
-        ...
+        """
+        Gives values of address balance.
+
+        Args:
+            address (`str`): Address string.
+
+        Returns:
+            int: value of balance at the address.
+        """
 
     @abstractmethod
     def get_code(self, address: str) -> bytes:
-        ...
+        """
+        Get the bytes of the contract.
+        Must override address or else raise `error`.
```

Comment:
The abstract method does not raise an error though?

Reply:
oh okay ill rm that line